// vscode-gradle/lib/proto/gradle_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var gradle_pb = require('./gradle_pb.js');

function serialize_gradle_CancelBuildReply(arg) {
  if (!(arg instanceof gradle_pb.CancelBuildReply)) {
    throw new Error('Expected argument of type gradle.CancelBuildReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelBuildReply(buffer_arg) {
  return gradle_pb.CancelBuildReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_CancelBuildRequest(arg) {
  if (!(arg instanceof gradle_pb.CancelBuildRequest)) {
    throw new Error('Expected argument of type gradle.CancelBuildRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelBuildRequest(buffer_arg) {
  return gradle_pb.CancelBuildRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_CancelBuildsReply(arg) {
  if (!(arg instanceof gradle_pb.CancelBuildsReply)) {
    throw new Error('Expected argument of type gradle.CancelBuildsReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelBuildsReply(buffer_arg) {
  return gradle_pb.CancelBuildsReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_CancelBuildsRequest(arg) {
  if (!(arg instanceof gradle_pb.CancelBuildsRequest)) {
    throw new Error('Expected argument of type gradle.CancelBuildsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelBuildsRequest(buffer_arg) {
  return gradle_pb.CancelBuildsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_CancelProjectsReply(arg) {
  if (!(arg instanceof gradle_pb.CancelProjectsReply)) {
    throw new Error('Expected argument of type gradle.CancelProjectsReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelProjectsReply(buffer_arg) {
  return gradle_pb.CancelProjectsReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_CancelProjectsRequest(arg) {
  if (!(arg instanceof gradle_pb.CancelProjectsRequest)) {
    throw new Error('Expected argument of type gradle.CancelProjectsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_CancelProjectsRequest(buffer_arg) {
  return gradle_pb.CancelProjectsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetBuildReply(arg) {
  if (!(arg instanceof gradle_pb.GetBuildReply)) {
    throw new Error('Expected argument of type gradle.GetBuildReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetBuildReply(buffer_arg) {
  return gradle_pb.GetBuildReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetBuildRequest(arg) {
  if (!(arg instanceof gradle_pb.GetBuildRequest)) {
    throw new Error('Expected argument of type gradle.GetBuildRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetBuildRequest(buffer_arg) {
  return gradle_pb.GetBuildRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetDaemonsStatusReply(arg) {
  if (!(arg instanceof gradle_pb.GetDaemonsStatusReply)) {
    throw new Error('Expected argument of type gradle.GetDaemonsStatusReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetDaemonsStatusReply(buffer_arg) {
  return gradle_pb.GetDaemonsStatusReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetDaemonsStatusRequest(arg) {
  if (!(arg instanceof gradle_pb.GetDaemonsStatusRequest)) {
    throw new Error('Expected argument of type gradle.GetDaemonsStatusRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetDaemonsStatusRequest(buffer_arg) {
  return gradle_pb.GetDaemonsStatusRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetProjectsReply(arg) {
  if (!(arg instanceof gradle_pb.GetProjectsReply)) {
    throw new Error('Expected argument of type gradle.GetProjectsReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetProjectsReply(buffer_arg) {
  return gradle_pb.GetProjectsReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_GetProjectsRequest(arg) {
  if (!(arg instanceof gradle_pb.GetProjectsRequest)) {
    throw new Error('Expected argument of type gradle.GetProjectsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_GetProjectsRequest(buffer_arg) {
  return gradle_pb.GetProjectsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_RunBuildReply(arg) {
  if (!(arg instanceof gradle_pb.RunBuildReply)) {
    throw new Error('Expected argument of type gradle.RunBuildReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_RunBuildReply(buffer_arg) {
  return gradle_pb.RunBuildReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_RunBuildRequest(arg) {
  if (!(arg instanceof gradle_pb.RunBuildRequest)) {
    throw new Error('Expected argument of type gradle.RunBuildRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_RunBuildRequest(buffer_arg) {
  return gradle_pb.RunBuildRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_StopDaemonReply(arg) {
  if (!(arg instanceof gradle_pb.StopDaemonReply)) {
    throw new Error('Expected argument of type gradle.StopDaemonReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_StopDaemonReply(buffer_arg) {
  return gradle_pb.StopDaemonReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_StopDaemonRequest(arg) {
  if (!(arg instanceof gradle_pb.StopDaemonRequest)) {
    throw new Error('Expected argument of type gradle.StopDaemonRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_StopDaemonRequest(buffer_arg) {
  return gradle_pb.StopDaemonRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_StopDaemonsReply(arg) {
  if (!(arg instanceof gradle_pb.StopDaemonsReply)) {
    throw new Error('Expected argument of type gradle.StopDaemonsReply');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_StopDaemonsReply(buffer_arg) {
  return gradle_pb.StopDaemonsReply.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_gradle_StopDaemonsRequest(arg) {
  if (!(arg instanceof gradle_pb.StopDaemonsRequest)) {
    throw new Error('Expected argument of type gradle.StopDaemonsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_gradle_StopDaemonsRequest(buffer_arg) {
  return gradle_pb.StopDaemonsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}


var GradleService = exports.GradleService = {
  getBuild: {
    path: '/gradle.Gradle/GetBuild',
    requestStream: false,
    responseStream: true,
    requestType: gradle_pb.GetBuildRequest,
    responseType: gradle_pb.GetBuildReply,
    requestSerialize: serialize_gradle_GetBuildRequest,
    requestDeserialize: deserialize_gradle_GetBuildRequest,
    responseSerialize: serialize_gradle_GetBuildReply,
    responseDeserialize: deserialize_gradle_GetBuildReply,
  },
  runBuild: {
    path: '/gradle.Gradle/RunBuild',
    requestStream: false,
    responseStream: true,
    requestType: gradle_pb.RunBuildRequest,
    responseType: gradle_pb.RunBuildReply,
    requestSerialize: serialize_gradle_RunBuildRequest,
    requestDeserialize: deserialize_gradle_RunBuildRequest,
    responseSerialize: serialize_gradle_RunBuildReply,
    responseDeserialize: deserialize_gradle_RunBuildReply,
  },
  getProjects: {
    path: '/gradle.Gradle/GetProjects',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.GetProjectsRequest,
    responseType: gradle_pb.GetProjectsReply,
    requestSerialize: serialize_gradle_GetProjectsRequest,
    requestDeserialize: deserialize_gradle_GetProjectsRequest,
    responseSerialize: serialize_gradle_GetProjectsReply,
    responseDeserialize: deserialize_gradle_GetProjectsReply,
  },
  cancelBuild: {
    path: '/gradle.Gradle/CancelBuild',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.CancelBuildRequest,
    responseType: gradle_pb.CancelBuildReply,
    requestSerialize: serialize_gradle_CancelBuildRequest,
    requestDeserialize: deserialize_gradle_CancelBuildRequest,
    responseSerialize: serialize_gradle_CancelBuildReply,
    responseDeserialize: deserialize_gradle_CancelBuildReply,
  },
  cancelBuilds: {
    path: '/gradle.Gradle/CancelBuilds',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.CancelBuildsRequest,
    responseType: gradle_pb.CancelBuildsReply,
    requestSerialize: serialize_gradle_CancelBuildsRequest,
    requestDeserialize: deserialize_gradle_CancelBuildsRequest,
    responseSerialize: serialize_gradle_CancelBuildsReply,
    responseDeserialize: deserialize_gradle_CancelBuildsReply,
  },
  cancelProjects: {
    path: '/gradle.Gradle/CancelProjects',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.CancelProjectsRequest,
    responseType: gradle_pb.CancelProjectsReply,
    requestSerialize: serialize_gradle_CancelProjectsRequest,
    requestDeserialize: deserialize_gradle_CancelProjectsRequest,
    responseSerialize: serialize_gradle_CancelProjectsReply,
    responseDeserialize: deserialize_gradle_CancelProjectsReply,
  },
  getDaemonsStatus: {
    path: '/gradle.Gradle/GetDaemonsStatus',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.GetDaemonsStatusRequest,
    responseType: gradle_pb.GetDaemonsStatusReply,
    requestSerialize: serialize_gradle_GetDaemonsStatusRequest,
    requestDeserialize: deserialize_gradle_GetDaemonsStatusRequest,
    responseSerialize: serialize_gradle_GetDaemonsStatusReply,
    responseDeserialize: deserialize_gradle_GetDaemonsStatusReply,
  },
  stopDaemons: {
    path: '/gradle.Gradle/StopDaemons',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.StopDaemonsRequest,
    responseType: gradle_pb.StopDaemonsReply,
    requestSerialize: serialize_gradle_StopDaemonsRequest,
    requestDeserialize: deserialize_gradle_StopDaemonsRequest,
    responseSerialize: serialize_gradle_StopDaemonsReply,
    responseDeserialize: deserialize_gradle_StopDaemonsReply,
  },
  stopDaemon: {
    path: '/gradle.Gradle/StopDaemon',
    requestStream: false,
    responseStream: false,
    requestType: gradle_pb.StopDaemonRequest,
    responseType: gradle_pb.StopDaemonReply,
    requestSerialize: serialize_gradle_StopDaemonRequest,
    requestDeserialize: deserialize_gradle_StopDaemonRequest,
    responseSerialize: serialize_gradle_StopDaemonReply,
    responseDeserialize: deserialize_gradle_StopDaemonReply,
  },
};

exports.GradleClient = grpc.makeGenericClientConstructor(GradleService);
